src/qtism/data/content/BodyElement.php

```diff
@@ -272,4 +329,290 @@ public function getDir()
+    {
+        return $this->ariaLabelledBy !== '';
+    }
+
+    /**
+     * @param string $ariaOwns
+     * @throws InvalidArgumentException
+     */
+    public function setAriaOwns($ariaOwns)
+    {
+        if ($ariaOwns !== '' && !Format::isAriaIdRefs($ariaOwns)) {
+            $val = (is_object($ariaOwns)) ? ('instance of ' . get_class($ariaOwns)) : $ariaOwns;
+
+            $msg = "'${val}' is not a valid value for attribute 'aria-owns'.";
+            throw new InvalidArgumentException($msg);
+        }
+
+        $this->ariaOwns = $ariaOwns;
+    }
+
+    /**
+     * @return string
+     */
+    public function getAriaOwns()
+    {
+        return $this->ariaOwns;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasAriaOwns()
+    {
+        return $this->ariaOwns !== '';
+    }
+
+    /**
+     * @param string $ariaLevel
+     * @throws InvalidArgumentException
+     */
+    public function setAriaLevel($ariaLevel)
+    {
+        if ($ariaLevel !== '' && !Format::isAriaLevel($ariaLevel)) {
+            $val = (is_object($ariaLevel)) ? ('instance of ' . get_class($ariaLevel)) : $ariaLevel;
+
+            $msg = "'${val}' is not a valid value for attribute 'aria-level'.";
+            throw new InvalidArgumentException($msg);
+        }
+
+        $this->ariaLevel = strval($ariaLevel);
+    }
+
+    /**
+     * @return string
+     */
+    public function getAriaLevel()
+    {
+        return $this->ariaLevel;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasAriaLevel()
+    {
+        return $this->ariaLevel !== '';
+    }
+
+    /**
+     * @param int $ariaLive A value from the AriaLive enumeration or false for no value.
+     * @throws InvalidArgumentException
+     */
+    public function setAriaLive($ariaLive)
+    {
+        if ($ariaLive === false || in_array($ariaLive, AriaLive::asArray(), true)) {
+            $this->ariaLive = $ariaLive;
+        } else {
+            $val = (is_object($ariaLive)) ? ('instance of ' . get_class($ariaLive)) : $ariaLive;
+            $msg = "'${val}' is not a valid value for attribute 'aria-live'.";
+            throw new InvalidArgumentException($msg);
+        }
+    }
+
+    /**
+     * @return int
+     */
+    public function getAriaLive()
+    {
+        return $this->ariaLive;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasAriaLive()
+    {
+        return $this->ariaLive !== false;
+    }
+
+    /**
+     * @param int $ariaOrientation A value from the AriaOrientation enumeration or false for no orientation.
+     * @throws InvalidArgumentException
+     */
+    public function setAriaOrientation($ariaOrientation)
+    {
+        if ($ariaOrientation === false || in_array($ariaOrientation, AriaOrientation::asArray(), true)) {
+            $this->ariaOrientation = $ariaOrientation;
+        } else {
+            $val = (is_object($ariaOrientation)) ? ('instance of ' . get_class($ariaOrientation)) : $ariaOrientation;
```

Comment:
it means that var could be an object?

Reply:
Potentially yes. Depends on the client code usage :smile: